Automation failure : test_positive_create_user_4_1

TestNames  : 
tests.foreman.cli.test_user.User.test_positive_create_user_4_1



please use valid email address for positive tests
Failed to create User with {u'firstname': u'rzHLHr75xi', u'lastname': u'owXoxzxecc', u'location-ids': None, u'password': u'7AMPHe2gtn', u'auth-source-id': 1, u'default-location-id': None, u'organization-ids': None, u'default-organization-id': None, u'admin': None, u'mail': u'\ua3a9\U000225cd\u5664\U00027a9e\U000204cc\U0002112b\U000250fa\U00027558\U00028403\U00024523@example.com', u'login': u'ZECIrl'} data due to:
  [ERROR 2015-05-06 17:52:40 Exception] Email address is invalid
  Could not create the user:
    Email address is invalid

$ hammer -v -u admin -p changeme --output=csv user create --firstname="rzHLHr75xi" --lastname="owXoxzxecc" --password="7AMPHe2gtn" --auth-source-id="1" --mail="ꎩ10𢗍噤10𢗍𧪞10𢗍𧪞𠓌10𢗍𧪞𠓌𡄫10𢗍𧪞𠓌𡄫𥃺10𢗍𧪞𠓌𡄫𥃺𧕘10𢗍𧪞𠓌𡄫𥃺𧕘𨐃10𢗍𧪞𠓌𡄫𥃺𧕘𨐃𤔣@example.com" --login="ZECIrl"
[ERROR 2015-05-07 08:02:51 Exception] Email address is invalid
Email address is too long (maximum is 60 characters)
Could not create the user:
  Email address is invalid
  Email address is too long (maximum is 60 characters)

please fix it
thanks

the result of test when executed locally using nosetests:

$ nosetests -c robottelo.properties -m  test_positive_create_user_4 /home/tkolhar/git/robottelo/tests/foreman/cli/test_user.py
/usr/lib/python2.7/site-packages/nose/plugins/manager.py:395: RuntimeWarning: Unable to load plugin xunitmp = nose_xunitmp:XunitMP: cannot import name force_unicode
  RuntimeWarning)
@Test: Create User for all variations of Email Address ... 2015-05-12 16:23:24 - robottelo - DEBUG - Running test User/test_positive_create_user_4_1__xec_xde_xde_xce_xf5_xd3_xe9_xd0_xeb_xe3_example_com
2015-05-12 16:23:24 - robottelo.cli.factory - DEBUG - User "r0qH1S" password not provided HV5X1HxdJd was generated
2015-05-12 16:23:24 - robottelo.common.ssh - DEBUG - >>> [qe-sat6-rhel71.usersys.redhat.com] LANG=en_US.UTF-8 hammer -v -u admin -p changeme --output=csv user create --firstname="P60vP1JvA4" --lastname="abOp66wUSE" --password="HV5X1HxdJd" --auth-source-id="1" --mail="ìÞÞÎõÓéÐëã@example.com" --login="r0qH1S"
2015-05-12 16:23:27 - robottelo.common.ssh - INFO - Instantiated Paramiko client 0x2924ed0
2015-05-12 16:23:29 - robottelo.common.ssh - INFO - Destroying Paramiko client 0x2924ed0
2015-05-12 16:23:29 - robottelo.common.ssh - INFO - Destroyed Paramiko client 0x2924ed0
2015-05-12 16:23:29 - robottelo.common.ssh - DEBUG - <<< [ERROR 2015-05-12 06:53:28 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

2015-05-12 16:23:29 - robottelo.cli.factory - DEBUG - [ERROR 2015-05-12 06:53:28 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

FAIL

======================================================================
FAIL: @Test: Create User for all variations of Email Address
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ddt.py", line 146, in wrapper
    return func(self, *args, **kwargs)
  File "/home/tkolhar/git/robottelo/tests/foreman/cli/test_user.py", line 215, in test_positive_create_user_4
    self.fail(err)
AssertionError: Failed to create User with {u'firstname': u'P60vP1JvA4', u'lastname': u'abOp66wUSE', u'location-ids': None, u'password': u'HV5X1HxdJd', u'auth-source-id': 1, u'default-location-id': None, u'organization-ids': None, u'default-organization-id': None, u'admin': None, u'mail': u'\xec\xde\xde\xce\xf5\xd3\xe9\xd0\xeb\xe3@example.com', u'login': u'r0qH1S'} data due to:
  [ERROR 2015-05-12 06:53:28 Exception] Email address is invalid
  Could not create the user:
    Email address is invalid
  

----------------------------------------------------------------------
XML: foreman-results.xml
----------------------------------------------------------------------
Ran 1 test in 4.176s

FAILED (failures=1)


Manual execution of test gives:

 $ hammer -v -u admin -p changeme --output=csv user create --firstname="P60vP1JvA4" --lastname="abOp66wUSE" --password="HV5X1HxdJd" --auth-source-id="1" --mail="ìÞÞÎõÓéÐëã@example.com" --login="r0qH1S"

[ERROR 2015-05-12 06:55:09 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

It seems that email consisting of latin alphabets are not accepted as valid email id
This is a  BZ as it should also accept latin alphabets as email addresses
https://bugzilla.redhat.com/show_bug.cgi?id=1204667
please add @skip_bug decorators until the bug gets fixed
this will allow the test to skip for a while

Better to run without the smoke flag:

```python
py.test tests/foreman/cli/test_user.py -ktest_positive_create_user_4
===================================== test session starts =====================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 202 items

tests/foreman/cli/test_user.py FF.......

========================================== FAILURES ===========================================
___ User.test_positive_create_user_4_1__xe3_xc9_xd9_xce_xd1_xfc_xc9_xcc_xdc_xf3_example_com ___

self = <tests.foreman.cli.test_user.User testMethod=test_positive_create_user_4_1__xe3_xc9_xd9_xce_xd1_xfc_xc9_xcc_xdc_xf3_example_com>

    @wraps(func)
    def wrapper(self):
>       return func(self, *args, **kwargs)

../../.virtualenvs/robottelo/lib/python2.7/site-packages/ddt.py:146:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/foreman/cli/test_user.py:215: in test_positive_create_user_4
    self.fail(err)
E   AssertionError: Failed to create User with {u'firstname': u'HhUHVGniuH', u'lastname': u'CzMdDhew9O', u'location-ids': None, u'password': u'9h5sTqhqcI', u'auth-source-id': 1, u'default-location-id': None, u'organization-ids': None, u'default-organization-id': None, u'admin': None, u'mail': u'\xe3\xc9\xd9\xce\xd1\xfc\xc9\xcc\xdc\xf3@example.com', u'login': u'VLihui'} data due to:
E     [ERROR 2015-05-12 08:31:44 Exception] Email address is invalid
E     Could not create the user:
E       Email address is invalid
------------------------------------ Captured stdout call -------------------------------------
2015-05-12 09:31:40 - robottelo - DEBUG - Running test User/test_positive_create_user_4_1__xe3_xc9_xd9_xce_xd1_xfc_xc9_xcc_xdc_xf3_example_com
2015-05-12 09:31:40 - robottelo.cli.factory - DEBUG - User "VLihui" password not provided 9h5sTqhqcI was generated
2015-05-12 09:31:40 - robottelo.common.ssh - DEBUG - >>> [rhsm-qe-4.rhq.lab.eng.bos.redhat.com] LANG=en_US.UTF-8 hammer -v -u admin -p changeme --output=csv user create --firstname="HhUHVGniuH" --lastname="CzMdDhew9O" --password="9h5sTqhqcI" --auth-source-id="1" --mail="ãÉÙÎÑüÉÌÜó@example.com" --login="VLihui"
2015-05-12 09:31:42 - robottelo.common.ssh - INFO - Instantiated Paramiko client 0x10ebb2290
2015-05-12 09:31:44 - robottelo.common.ssh - INFO - Destroying Paramiko client 0x10ebb2290
2015-05-12 09:31:44 - robottelo.common.ssh - INFO - Destroyed Paramiko client 0x10ebb2290
2015-05-12 09:31:44 - robottelo.common.ssh - DEBUG - <<< [ERROR 2015-05-12 08:31:44 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

2015-05-12 09:31:44 - robottelo.cli.factory - DEBUG - [ERROR 2015-05-12 08:31:44 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

 User.test_positive_create_user_4_2__ubf8e_u7588_u890c_ufc61_u3817_u6f80_u603d_ub5a6_u42ab_u860b_example_com

self = <tests.foreman.cli.test_user.User testMethod=test_positive_create_user_4_2__ubf8e_u7588_u890c_ufc61_u3817_u6f80_u603d_ub5a6_u42ab_u860b_example_com>

    @wraps(func)
    def wrapper(self):
>       return func(self, *args, **kwargs)

../../.virtualenvs/robottelo/lib/python2.7/site-packages/ddt.py:146:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/foreman/cli/test_user.py:215: in test_positive_create_user_4
    self.fail(err)
E   AssertionError: Failed to create User with {u'firstname': u'6qDQ1992Hi', u'lastname': u'nQ1JEfdX0w', u'location-ids': None, u'password': u'HL7ojH5F1v', u'auth-source-id': 1, u'default-location-id': None, u'organization-ids': None, u'default-organization-id': None, u'admin': None, u'mail': u'\ubf8e\u7588\u890c\ufc61\u3817\u6f80\u603d\ub5a6\u42ab\u860b@example.com', u'login': u'5Gp9f4'} data due to:
E     [ERROR 2015-05-12 08:31:47 Exception] Email address is invalid
E     Could not create the user:
E       Email address is invalid
------------------------------------ Captured stdout call -------------------------------------
2015-05-12 09:31:44 - robottelo - DEBUG - Running test User/test_positive_create_user_4_2__ubf8e_u7588_u890c_ufc61_u3817_u6f80_u603d_ub5a6_u42ab_u860b_example_com
2015-05-12 09:31:44 - robottelo.cli.factory - DEBUG - User "5Gp9f4" password not provided HL7ojH5F1v was generated
2015-05-12 09:31:44 - robottelo.common.ssh - DEBUG - >>> [rhsm-qe-4.rhq.lab.eng.bos.redhat.com] LANG=en_US.UTF-8 hammer -v -u admin -p changeme --output=csv user create --firstname="6qDQ1992Hi" --lastname="nQ1JEfdX0w" --password="HL7ojH5F1v" --auth-source-id="1" --mail="뾎疈褌ﱡ㠗澀怽떦䊫蘋@example.com" --login="5Gp9f4"
2015-05-12 09:31:45 - robottelo.common.ssh - INFO - Instantiated Paramiko client 0x10ebb8710
2015-05-12 09:31:47 - robottelo.common.ssh - INFO - Destroying Paramiko client 0x10ebb8710
2015-05-12 09:31:47 - robottelo.common.ssh - INFO - Destroyed Paramiko client 0x10ebb8710
2015-05-12 09:31:47 - robottelo.common.ssh - DEBUG - <<< [ERROR 2015-05-12 08:31:47 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

2015-05-12 09:31:47 - robottelo.cli.factory - DEBUG - [ERROR 2015-05-12 08:31:47 Exception] Email address is invalid
Could not create the user:
  Email address is invalid

=================== 193 tests deselected by '-ktest_positive_create_user_4' ===================
===================== 2 failed, 7 passed, 193 deselected in 77.73 seconds =====================
```

--- a/tests/foreman/cli/test_user.py
+++ b/tests/foreman/cli/test_user.py
@@ -187,17 +187,16 @@ class User(CLITestCase):
         self.__assert_exists(args)
 
     @data(
-        u'{0}@example.com'.format(gen_string("latin1")),
-        u'{0}@example.com'.format(gen_string("utf8")),
-        u'{0}@example.com'.format(gen_string("alpha")),
-        u'{0}@example.com'.format(gen_string("alphanumeric")),
-        u'{0}@example.com'.format(gen_string("numeric")),
-        u'{0}@example.com'.format(gen_string("alphanumeric", 48)),
+        u'{0}@example.com'.format(gen_string('alpha')),
+        u'{0}@example.com'.format(gen_string('alphanumeric')),
+        u'{0}@example.com'.format(gen_string('numeric')),
+        u'{0}@example.com'.format(gen_string('alphanumeric', 48)),
         u'{0}+{1}@example.com'.format(gen_alphanumeric(), gen_alphanumeric()),
         u'{0}.{1}@example.com'.format(gen_alphanumeric(), gen_alphanumeric()),
-        r"!#$%&*+-/=?^\`{|}~@example.com",
+        u'"():;"@example.com',
+        r'!#$%&*+-/=?^`{|}~@example.com',
     )
-    def test_positive_create_user_4(self, test_data):
+    def test_positive_create_user_4(self, email):
         """@Test: Create User for all variations of Email Address
 
         @Feature: User - Positive Create
@@ -209,11 +208,14 @@ class User(CLITestCase):
         @Assert: User is created
 
         """
+        # The email must be escaped because some characters to not fail the
+        # parsing of the generated shell command
+        escaped_email = email.replace('"', r'\"').replace('`', r'\`')
         try:
-            args = make_user({'mail': test_data})
+            user = make_user({'mail': escaped_email})
         except CLIFactoryError as err:
             self.fail(err)
-        self.__assert_exists(args)
+        self.assertEqual(user['email'], email)
 
     @data({'password': gen_string("latin1")},
           {'password': gen_string("utf8")},

